plugins/woo-ai/src/hooks/useProductDataSuggestions.ts

```diff
@@ -24,14 +25,15 @@ export const useProductDataSuggestions = () => {
 		request: ProductDataSuggestionRequest
 	): Promise< ProductDataSuggestion[] > => {
 		try {
-			const response =
-				await apiFetch< ProductDataSuggestionSuccessResponse >( {
+			const response = await requestJetpackToken().then( ( { token } ) =>
+				apiFetch< ProductDataSuggestionSuccessResponse >( {
 					path: '/wooai/product-data-suggestions',
 					method: 'POST',
-					data: request,
-				} );
+					data: { ...request, token },
+				} )
+			);
 
-			return response.suggestions;
+			return JSON.parse( response.completion ).suggestions;
```

Comment:
This JSON parsing should likely be done server-side, but I was having trouble getting this done since my logging isn't working within that code.